File: src/protocol/AuthenticateResponse.cpp
#include "AuthenticateResponse.h"
#include "ResponseVisitor.h"

using namespace std;
using namespace communication;

AuthenticateResponse::AuthenticateResponse()
        : Response(10, ResponseType::AUTHENTICATE)
{}

AuthenticateResponse::~AuthenticateResponse()
{}

vector<uint8_t> AuthenticateResponse::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(responseType_));

    return frame;
}

string AuthenticateResponse::getName()
{
    return string("AuthenticateResponse");
}

void AuthenticateResponse::accept(ResponseVisitor &visitor)
{
    visitor.visit(*this);
}

void AuthenticateResponse::initializeDataSize()
{
    uint16_t dataSize = sizeof(responseType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/protocol/AuthenticateResponse.java
package protocol;

import java.util.ArrayList;

public final class AuthenticateResponse extends Response
	{
		public AuthenticateResponse()
		{
			super(10, ResponseType.AUTHENTICATE);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(responseType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("AuthenticateResponse");
		}
		@Override
		public void accept(ResponseVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}